.github/workflows/tf-aws-test.yml

```diff
@@ -3,10 +3,16 @@ on:
   release:
     types:
       - published
+  push:
+    branches:
+      - main
```

Comment:
In order for this workflow to block the build, you would need to do one of these things:

1. Move this workflow to be a job of the build.yml workflow, making it run in run similar to hangar
2. Make the build workflow wait for this to succeed, probably by checking the github API or finding an action that handles that

(1) is much easier. The problem is that to run it before the build means you cannot install wing normally, so you'll have to use the same mechanism that hangar uses.